genai/example_test.go

```diff
@@ -252,6 +252,26 @@ func ExampleEmbeddingModel_EmbedContent() {
 	fmt.Println(res.Embedding.Values)
 }
 
+func ExampleEmbeddingBatch() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("API_KEY")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+	em := client.EmbeddingModel("embedding-001")
+	b := em.NewBatch().
```

Comment:
I'm a bit bothered by `EmbedContents` living on `EmbeddingBatch`. It feels like `EmbeddingBatch` is here just to help create a list of embedding requests. One option could be for `EmbeddingModel` to have an `EmbedBatch` method that takes an `EmbeddingBatch`

... or even an `EmbedBatch` method that just takes parts, wraps them in request, and sends them out? Won't this work?

An alternative API could be that

Reply:
> I'm a bit bothered by EmbedContents living on EmbeddingBatch

It is a bit cute. It does let you chain nicely:
```
res, err := m.NewBatch().
   Add(...).
   Add(...).
   EmbedContents(...)
```

There is precedent in the Cloud Client Libraries:

- https://pkg.go.dev/cloud.google.com/go/firestore#BulkWriter
- https://pkg.go.dev/cloud.google.com/go/spanner#BatchReadOnlyTransaction

But happy switch to 
```
func (*EmbeddingModel) EmbedContents(ctx, *EmbeddingBatch)
```
instead if you prefer.

> an EmbedBatch method that just takes parts, wraps them in request, and sends them out? Won't this work?

That's where the question about title comes in. If each content could have a different title, then the signature of that function is problematic.